rpc/commands/board.proto

```diff
@@ -206,6 +206,22 @@ message BoardListAllResp {
     repeated BoardListItem boards = 1;
 }
 
+message BoardListWatchReq {
+    // Arduino Core Service instance from the `Init` response.
+    Instance instance = 1;
+    // Set this to true to stop the discovery process
+    bool interrupt = 2;
+}
+
+message BoardListWatchResp {
+    // Event type as received from the serial discovery tool
+    string event_type = 1;
+    // Information about the port
+    DetectedPort port = 2;
+    // Eventual errors when detecting connected boards
+    string error = 3;
```

Comment:
I am not a `proto` expert and I hope @cmaglie can help, but I would have expected to have something like:
```proto
message BoardListWatchResp {
  oneof content {
    DetectedChange change = 1;
    string error = 2;
  }
}

message DetectedChange {
  enum ChangeType { ADDED = 0; REMOVED = 0; }
  ChangeType type = 1;
  DetectedPort port = 2;
}
```

Reply:
Neither am I and that's why didn't about this approach, I like it though.

Reply:
I'm positive to move the pair (ChangeType, DetectedPort) into their own separate message.
I would change the naming, since we are now talking about `Ports` IMHO it should be:

```
message BoardListWatchResp {
  oneof content {
    DetectedPortEvent event = 1;
    string error = 2;
  }
}

message DetectedPortEvent {
  enum Event { ADDED = 0; REMOVED = 0; }
  Event action = 1;
  DetectedPort port = 2;
}
```
(and at this point also we should probably go all-in and call it Port instead of DetectedPort, this is something that we must face once Pluggable Discovery is merged...)

I'm not sure about the `oneof`, there are other places where it may be useful to make the API more explicit, anyway when I tried it, IIRC, the generated API turned out to be very messy especially in languages without sum types like go, java, c++ where accessors (boilerplate) methods are added to handle it correctly. Let's evaluate if the pro (a cleaner `proto` API) is more valuable than the cons (a bigger "cognitive" load in the implementation).

Reply:
Thank you for your recommendations, @cmaglie. As I said, I am fine with this API now. It's your call whether you want to make the changes now or later.